common/helpers/picasso.js

```diff
@@ -214,3 +218,118 @@ function optimalGetUpTime(gap, scheduleEntryBounds, timeBucketSize) {
   // There is already a large enough margin, or there are no schedule entries ending around the getUpTime.
   return getUpTime
 }
+
+/**
+ * Generates an array of time row descriptions, used for labeling the vertical axis of the picasso.
+ * Format of each array element: [hour, weight] where weight determines how tall the time row is rendered.
+ *
+ * @returns {*[[hour: number, weight: number]]}
+ */
+export function times(getUpTime, bedTime, timeStep) {
+  const times = [[getUpTime - timeStep / 2, 0.5]]
+  for (let i = 0; getUpTime + i * timeStep < bedTime; i++) {
+    // TODO The weight could also be generated depending on the schedule entries present in the camp:
+    //   e.g. give less weight to hours that contain no schedule entries.
+    const weight = 1
+    times.push([getUpTime + i * timeStep, weight])
+  }
+  times.push([bedTime, 0.5])
+  // this last hour is only needed for defining the length of the day. The weight should be 0.
+  times.push([bedTime + timeStep / 2, 0])
+
+  return times
+}
+
+/**
+ * Returns the total sum of weights in the times array (with times format as generated by times() function)
+ */
+export function timesWeightsSum(times) {
+  return times.reduce((sum, [_, weight]) => sum + weight, 0)
+}
```

Comment:
Where is this from? Can this be replaced there as well?

Reply:
`times` I forgot to remove, fixed with https://github.com/ecamp/ecamp3/pull/3887/commits/ba5eaba2703db4df5cecc5f00e9c5b0a7a7112e8

timesWeightsSum is coming from this line: https://github.com/ecamp/ecamp3/pull/3887/files#diff-f31cadd401f377a07346097a1a050c7f5fd49e7e7d7b98cca969ab2af6e9b57aL144